Bottom: cf7d7cbb9f69ae3049bf5adbe7d509741569e657
Top:    614388263ecf20dae3067e91acbefa866f088212
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 16:18:35 +0100

Refresh of draw-c-never-call-drawpinorvia

---

diff --git a/src/draw.c b/src/draw.c
index 13fdd5c..6672448 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -34,6 +34,8 @@
 #include "config.h"
 #endif
 
+#include <assert.h>
+
 #include "global.h"
 
 /*#include "clip.h"*/
@@ -1171,16 +1173,9 @@ DrawElementPackageLowLevel (ElementTypePtr Element)
 void
 DrawVia (PinTypePtr Via)
 {
-  if (Gathering)
-    {
-      AddPart (pv);
-      return;
-    }
-  else
-    {
-      SetPVColor (Via, VIA_TYPE);
-      DrawPinOrViaLowLevel (Via, true);
-    }
+  assert (Gathering);
+
+  AddPart (Via);
 
   if (!TEST_FLAG (HOLEFLAG, Via) && TEST_FLAG (DISPLAYNAMEFLAG, Via))
     DrawPinOrViaNameLowLevel (Via);
@@ -1208,16 +1203,9 @@ DrawViaName (PinTypePtr Via)
 void
 DrawPin (PinTypePtr Pin)
 {
-  if (Gathering)
-    {
-      AddPart (Pin);
-      return;
-    }
-  else
-    {
-      SetPVColor (Pin, PIN_TYPE);
-      DrawPinOrViaLowLevel (Pin, true);
-    }
+  assert (Gathering);
+
+  AddPart (Pin);
 
   if ((!TEST_FLAG (HOLEFLAG, Pin) && TEST_FLAG (DISPLAYNAMEFLAG, Pin))
       || doing_pinout)
